x-pack/plugins/cloud_integrations/cloud_data_migration/server/config.ts

```diff
@@ -9,7 +9,12 @@ import { schema, TypeOf } from '@kbn/config-schema';
 import { PluginConfigDescriptor } from '@kbn/core-plugins-server';
 
 const configSchema = schema.object({
-  enabled: schema.boolean({ defaultValue: true }),
+  enabled: schema.conditional(
+    schema.contextRef('serverless'),
+    true,
+    schema.boolean({ defaultValue: true }),
```

Comment:
By taking this approach, we also set the setting to `false` in the `serverless.yml` file. This would give us a clear view of all of the plugins that are disabled. Alternatively, we can set the default value to `false` here and then remove the configuration from the `serverless.yml` file.

Reply:
++ I think the approach of explicitly setting `false` in the serverless.yml makes things easier to reason about and prevents our serverless configs from spreading out into more places, even if it seems counterintuitive to default to `true` here. You could also consider adding a comment to these to make that even clearer for folks in the future.

Reply:
Thanks! Good idea about adding a comment 👍